diff -up cp2k-2.7.0/arch/Linux-x86-64-gfortran.popt.shared cp2k-2.7.0/arch/Linux-x86-64-gfortran.popt
--- cp2k-2.7.0/arch/Linux-x86-64-gfortran.popt.shared	2015-09-25 13:03:32.227945728 +0200
+++ cp2k-2.7.0/arch/Linux-x86-64-gfortran.popt	2015-09-25 13:03:32.240945789 +0200
@@ -3,6 +3,7 @@ CC         = gcc
 CPP        =
 FC         = mpifort
 LD         = mpifort
+LD_SHARED  = mpifort -shared
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
diff -up cp2k-2.7.0/arch/Linux-x86-64-gfortran.psmp.shared cp2k-2.7.0/arch/Linux-x86-64-gfortran.psmp
--- cp2k-2.7.0/arch/Linux-x86-64-gfortran.psmp.shared	2015-09-25 13:03:32.228945733 +0200
+++ cp2k-2.7.0/arch/Linux-x86-64-gfortran.psmp	2015-09-25 13:03:32.692947903 +0200
@@ -3,6 +3,7 @@ CC         = gcc
 CPP        =
 FC         = mpifort
 LD         = mpifort
+LD_SHARED  = mpifort -shared
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
diff -up cp2k-2.7.0/arch/Linux-x86-64-gfortran.sopt.shared cp2k-2.7.0/arch/Linux-x86-64-gfortran.sopt
--- cp2k-2.7.0/arch/Linux-x86-64-gfortran.sopt.shared	2015-09-25 13:03:32.229945738 +0200
+++ cp2k-2.7.0/arch/Linux-x86-64-gfortran.sopt	2015-09-25 13:34:14.019818093 +0200
@@ -3,6 +3,7 @@ CC         = gcc
 CPP        =
 FC         = gfortran
 LD         = gfortran
+LD_SHARED  = gfortran -shared
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
diff -up cp2k-2.7.0/arch/Linux-x86-64-gfortran.ssmp.shared cp2k-2.7.0/arch/Linux-x86-64-gfortran.ssmp
--- cp2k-2.7.0/arch/Linux-x86-64-gfortran.ssmp.shared	2015-09-25 13:03:32.230945742 +0200
+++ cp2k-2.7.0/arch/Linux-x86-64-gfortran.ssmp	2015-09-25 13:03:32.697947926 +0200
@@ -3,6 +3,7 @@ CC         = gcc
 CPP        =
 FC         = gfortran
 LD         = gfortran
+LD_SHARED  = gfortran -shared
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
