diff -up cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/arch/Linux-x86-64-gfortran.psmp.r cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/arch/Linux-x86-64-gfortran.psmp
--- cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/arch/Linux-x86-64-gfortran.psmp.r	2020-09-25 17:01:47.000000000 +0200
+++ cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/arch/Linux-x86-64-gfortran.psmp	2020-09-29 15:18:55.464317641 +0200
@@ -5,64 +5,37 @@
 CC          = mpicc
 FC          = mpif90
 LD          = mpif90
+LD_SHARED   = mpif90 -shared
 AR          = ar -r
 
-include       $(MPI_PATH)/plumed/2.5.2/lib/plumed/src/lib/Plumed.inc.static
-
-ELPA_VER    = 2019.05.001
-ELPA_INC    = $(MPI_PATH)/elpa/$(ELPA_VER)-openmp/include/elpa_openmp-$(ELPA_VER)
-ELPA_LIB    = $(MPI_PATH)/elpa/$(ELPA_VER)-openmp/lib
-
-FFTW_INC    = $(GNU_PATH)/fftw/3.3/include
-FFTW_LIB    = $(GNU_PATH)/fftw/3.3/lib
-
-LIBINT_INC  = $(GNU_PATH)/libint/2.6.0-lmax-6/include
-LIBINT_LIB  = $(GNU_PATH)/libint/2.6.0-lmax-6/lib
-
-LIBXC_INC   = $(GNU_PATH)/libxc/4.3.4/include
-LIBXC_LIB   = $(GNU_PATH)/libxc/4.3.4/lib
-
-LIBXSMM_INC = $(GNU_PATH)/libxsmm/1.15/include
-LIBXSMM_LIB = $(GNU_PATH)/libxsmm/1.15/lib
-
-SPGLIB_INC  = $(GNU_PATH)/spglib/1.12.2/include
-SPGLIB_LIB  = $(GNU_PATH)/spglib/1.12.2/lib
-
-CFLAGS      = -O2 -fopenmp -g -mtune=native
+OPTFLAGS    = -O2 -g -mtune=native
+CFLAGS      = $(OPTFLAGS) -fopenmp
 
 DFLAGS      = -D__ELPA
 DFLAGS     += -D__FFTW3
 DFLAGS     += -D__LIBINT
 DFLAGS     += -D__LIBXC
-DFLAGS     += -D__LIBXSMM
+DFLAGS     += @LIBSMM_DEFS@
 DFLAGS     += -D__MPI_VERSION=3
-DFLAGS     += -D__PLUMED2
-DFLAGS     += -D__SPGLIB
+DFLAGS     += -D__MAX_CONTR=4
+DFLAGS     += -D__HAS_IEEE_EXCEPTIONS
 DFLAGS     += -D__parallel
 DFLAGS     += -D__SCALAPACK
 
 FCFLAGS     = $(CFLAGS) $(DFLAGS)
 FCFLAGS    += -ffree-form
 FCFLAGS    += -ffree-line-length-none
-FCFLAGS    += -ftree-vectorize
-FCFLAGS    += -funroll-loops
 FCFLAGS    += -std=f2008
-FCFLAGS    += -I$(ELPA_INC)/elpa -I$(ELPA_INC)/modules
-FCFLAGS    += -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC) -I$(LIBXSMM_INC)
 
-LDFLAGS     = $(FCFLAGS) -static-libgfortran
+LDFLAGS     = $(FCFLAGS) $(DISTLDFLAGS)
 
-LIBS        = $(PLUMED_DEPENDENCIES) -lgsl -lgslcblas -lz
-LIBS       += $(ELPA_LIB)/libelpa_openmp.a
-LIBS       += $(LIBXC_LIB)/libxcf03.a
-LIBS       += $(LIBXC_LIB)/libxc.a
-LIBS       += $(LIBINT_LIB)/libint2.a
-LIBS       += $(SPGLIB_LIB)/libsymspg.a
-LIBS       += $(FFTW_LIB)/libfftw3.a
-LIBS       += $(FFTW_LIB)/libfftw3_threads.a
-LIBS       += $(LIBXSMM_LIB)/libxsmmf.a
-LIBS       += $(LIBXSMM_LIB)/libxsmm.a
-LIBS       += $(MPI_LIBRARY_PATH)/libscalapack.a
-LIBS       += $(LIBPATH)/liblapack.a
-LIBS       += $(LIBPATH)/libblas.a
+LIBS        = -lelpa_openmp
+LIBS       += -lxcf90
+LIBS       += -lxc
+LIBS       += -lint2
+LIBS       += -lfftw3
+LIBS       += -lfftw3_omp
+LIBS       += @LIBSMM_LIBS@
+LIBS       += -lscalapack
+LIBS       += -lflexiblas
 LIBS       += -ldl -lpthread -lstdc++
diff -up cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/arch/Linux-x86-64-gfortran.ssmp.r cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/arch/Linux-x86-64-gfortran.ssmp
--- cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/arch/Linux-x86-64-gfortran.ssmp.r	2020-09-25 17:01:47.000000000 +0200
+++ cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/arch/Linux-x86-64-gfortran.ssmp	2020-09-29 15:19:10.681243626 +0200
@@ -4,49 +4,31 @@
 CC          = gcc
 FC          = gfortran
 LD          = gfortran
+LD_SHARED   = gfortran -shared
 AR          = ar -r
 
-FFTW_INC    = $(GNU_PATH)/fftw/3.3/include
-FFTW_LIB    = $(GNU_PATH)/fftw/3.3/lib
-
-LIBINT_INC  = $(GNU_PATH)/libint/2.6.0-lmax-6/include
-LIBINT_LIB  = $(GNU_PATH)/libint/2.6.0-lmax-6/lib
-
-LIBXC_INC   = $(GNU_PATH)/libxc/4.3.4/include
-LIBXC_LIB   = $(GNU_PATH)/libxc/4.3.4/lib
-
-LIBXSMM_INC = $(GNU_PATH)/libxsmm/1.15/include
-LIBXSMM_LIB = $(GNU_PATH)/libxsmm/1.15/lib
-
-SPGLIB_INC  = $(GNU_PATH)/spglib/1.12.2/include
-SPGLIB_LIB  = $(GNU_PATH)/spglib/1.12.2/lib
-
-CFLAGS      = -O2 -fopenmp -g -mtune=native
+OPTFLAGS    = -O2 -g -mtune=native
+CFLAGS      = $(OPTFLAGS) -fopenmp
 
 DFLAGS      = -D__FFTW3
 DFLAGS     += -D__LIBINT
 DFLAGS     += -D__LIBXC
-DFLAGS     += -D__LIBXSMM
-DFLAGS     += -D__SPGLIB
+DFLAGS     += @LIBSMM_DEFS@
+DFLAGS     += -D__HAS_IEEE_EXCEPTIONS
+DFLAGS     += -D__MAX_CONTR=4
 
 FCFLAGS     = $(CFLAGS) $(DFLAGS)
 FCFLAGS    += -ffree-form
 FCFLAGS    += -ffree-line-length-none
-FCFLAGS    += -ftree-vectorize
-FCFLAGS    += -funroll-loops
 FCFLAGS    += -std=f2008
-FCFLAGS    += -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC) -I$(LIBXSMM_INC)
 
-LDFLAGS     = $(FCFLAGS) -static
+LDFLAGS     = $(FCFLAGS) $(DISTLDFLAGS)
 
-LIBS        = $(LIBXC_LIB)/libxcf03.a
-LIBS       += $(LIBXC_LIB)/libxc.a
-LIBS       += $(LIBINT_LIB)/libint2.a
-LIBS       += $(SPGLIB_LIB)/libsymspg.a
-LIBS       += $(FFTW_LIB)/libfftw3.a
-LIBS       += $(FFTW_LIB)/libfftw3_threads.a
-LIBS       += $(LIBXSMM_LIB)/libxsmmf.a
-LIBS       += $(LIBXSMM_LIB)/libxsmm.a
-LIBS       += $(LIBPATH)/liblapack.a
-LIBS       += $(LIBPATH)/libblas.a
+LIBS        = -lxcf90
+LIBS       += -lxc
+LIBS       += -lint2
+LIBS       += -lfftw3
+LIBS       += -lfftw3_omp
+LIBS       += @LIBSMM_LIBS@
+LIBS       += -lflexiblas
 LIBS       += -ldl -lpthread -lstdc++
diff -up cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/exts/dbcsr/.cp2k/Makefile.r cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/exts/dbcsr/.cp2k/Makefile
--- cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/exts/dbcsr/.cp2k/Makefile.r	2020-09-13 19:31:19.000000000 +0200
+++ cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/exts/dbcsr/.cp2k/Makefile	2020-09-29 11:58:26.238474915 +0200
@@ -54,7 +54,7 @@ MAKEFILE     := $(DBCSRCP2K)/Makefile
 LIBDIR       := $(DBCSRHOME)/lib
 OBJDIR       := $(DBCSRHOME)/obj
 TOOLSDIR     := $(DBCSRHOME)/tools
-FYPPEXE      := $(TOOLSDIR)/build_utils/fypp/bin/fypp
+FYPPEXE      := fypp
 SRCDIR       := $(DBCSRHOME)/src
 TESTSDIR     := $(DBCSRHOME)/tests
 INCLUDEMAKE  :=
diff -up cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/Makefile.r cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/Makefile
--- cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/Makefile.r	2020-09-25 17:01:47.000000000 +0200
+++ cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/Makefile	2020-09-29 11:58:26.243474891 +0200
@@ -16,7 +16,7 @@ export VERSION=ssmp
 
 MAKEFILE     := $(CP2KHOME)/Makefile
 ARCHDIR      := $(CP2KHOME)/arch
-DATA_DIR     := $(CP2KHOME)/data
+DATA_DIR     := /usr/share/cp2k/data
 MAINEXEDIR   := $(CP2KHOME)/exe
 MAINLIBDIR   := $(CP2KHOME)/lib
 MAINOBJDIR   := $(CP2KHOME)/obj
@@ -501,7 +501,7 @@ FCFLAGS += -D__COMPILE_ARCH="\"$(ARCH)\"
 FYPPFLAGS ?= -n
 
 %.o: %.F
-	$(TOOLSRC)/build_utils/fypp $(FYPPFLAGS) $< $*.F90
+	fypp $(FYPPFLAGS) $< $*.F90
 	$(FC) -c $(FCFLAGS) -D__SHORT_FILE__="\"$(subst $(SRCDIR)/,,$<)\"" -I'$(dir $<)' $(OBJEXTSINCL) $*.F90 $(FCLOGPIPE)
 
 %.o: %.c
diff -up cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/tools/regtesting/do_regtest.r cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/tools/regtesting/do_regtest
--- cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/tools/regtesting/do_regtest.r	2020-09-25 17:01:47.000000000 +0200
+++ cp2k-dbf7a770d1541ba72a4652ee218de80c0484db2d/tools/regtesting/do_regtest	2020-09-29 13:59:45.419222477 +0200
@@ -383,6 +383,7 @@ function run_regtest_dir() {
      test_tolerance=`grep '^\s*\w' TEST_FILES | ${awk} -v itest=$itest -v def_err_tol=$default_err_tolerance '{c=c+1;if (c==itest) if (NF >= 3) { print $3 } else { print def_err_tol } }'`
      test_ref_value=`grep '^\s*\w' TEST_FILES | ${awk} -v itest=$itest '{c=c+1;if (c==itest) if (NF == 4) { print $4 } }'`
 
+     echo "Running ${input_file}"
      output_file=${dir_test}/${dir}/${input_file}.out
      output_last=${dir_last}/${dir}/${input_file}.out
 
@@ -1201,7 +1202,7 @@ EOF
    if (( n_leaks > 0 )); then
        REPORT_SUMMARY+=$(printf "; memleaks: %d" ${n_leaks})
    fi
-   REPORT_SUMMARY+=$(python -c "print('; %.0fmin'%(${full_timing_all}/60.0))")
+   REPORT_SUMMARY+=$(python3 -c "print('; %.0fmin'%(${full_timing_all}/60.0))")
    echo -e "\nSummary: ${REPORT_SUMMARY}"
 
    if ((n_wrong_results > 0)) || ((n_runtime_error > 0)) || ((n_leaks > 0)); then
