diff -up cp2k-6.1/arch/Linux-x86-64-gfortran.popt.r cp2k-6.1/arch/Linux-x86-64-gfortran.popt
--- cp2k-6.1/arch/Linux-x86-64-gfortran.popt.r	2018-06-12 14:07:45.000000000 +0200
+++ cp2k-6.1/arch/Linux-x86-64-gfortran.popt	2018-07-16 15:44:30.013193276 +0200
@@ -2,8 +2,9 @@
 # Author: Matthias Krack (matthias.krack@psi.ch, PSI, June 2018)
 CC          = gcc
 CPP         =
-FC          = mpif90
-LD          = mpif90
+FC          = mpifort
+LD          = mpifort
+LD_SHARED   = mpifort -shared
 AR          = ar -r
 FFTW_INC    = $(GCC_DIR)/fftw/3.3/include
 FFTW_LIB    = $(GCC_DIR)/fftw/3.3/lib
@@ -13,22 +14,26 @@ LIBXC_INC   = $(GCC_DIR)/libxc/4.0.4/inc
 LIBXC_LIB   = $(GCC_DIR)/libxc/4.0.4/lib
 LIBXSMM_INC = $(GCC_DIR)/libxsmm/1.9/include
 LIBXSMM_LIB = $(GCC_DIR)/libxsmm/1.9/lib
-DFLAGS      = -D__F2008 -D__FFTW3 -D__LIBINT -D__LIBXC -D__LIBXSMM -D__MPI_VERSION=3\
-              -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4\
+DFLAGS      = -D__F2008 -D__FFTW3 -D__LIBINT -D__LIBXC @LIBSMM_DEFS@ -D__MPI_VERSION=3\
+              -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4\
+              -D__SCALAPACK2 -D__ELPA=201705 -D__HAS_IEEE_EXCEPTIONS\
               -D__parallel -D__SCALAPACK
 CPPFLAGS    =
-FCFLAGS     = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
-              -ftree-vectorize -funroll-loops -mtune=native -std=f2008\
+OPTFLAGS    = -O2 -ffast-math\
+              -ftree-vectorize -funroll-loops -mtune=native\
               -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC) -I$(LIBXSMM_INC)
-LDFLAGS     = $(FCFLAGS) -static
-LIBS        = $(MPI_LIBRARY_PATH)/libscalapack-gnu.a\
-              $(LIBPATH)/liblapack-gnu.a\
-              $(LIBPATH)/libblas-gnu.a\
-              $(FFTW_LIB)/libfftw3.a\
-              $(LIBXC_LIB)/libxcf03.a\
-              $(LIBXC_LIB)/libxc.a\
-              $(LIBINT_LIB)/libderiv.a\
-              $(LIBINT_LIB)/libint.a\
-              $(LIBXSMM_LIB)/libxsmmf.a\
-              $(LIBXSMM_LIB)/libxsmm.a\
+CFLAGS      = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS    = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS     = $(DFLAGS) $(OPTFLAGS) -ffree-form -ffree-line-length-none -std=f2008
+LDFLAGS     = $(FCFLAGS) $(DISTLDFLAGS)
+LIBS        = -lelpa\
+              -lscalapack -lstdc++\
+              -lopenblas\
+              -lfftw3\
+              -lxcf03\
+              -lxc\
+              -lderiv\
+              -lint\
+              @LIBSMM_LIBS@\
               -ldl
+DATA_DIR    = /usr/share/cp2k
diff -up cp2k-6.1/arch/Linux-x86-64-gfortran.psmp.r cp2k-6.1/arch/Linux-x86-64-gfortran.psmp
--- cp2k-6.1/arch/Linux-x86-64-gfortran.psmp.r	2018-06-12 14:07:45.000000000 +0200
+++ cp2k-6.1/arch/Linux-x86-64-gfortran.psmp	2018-07-16 15:44:20.853109669 +0200
@@ -2,8 +2,9 @@
 # Author: Matthias Krack (matthias.krack@psi.ch, PSI, June 2018)
 CC          = gcc
 CPP         =
-FC          = mpif90
-LD          = mpif90
+FC          = mpifort
+LD          = mpifort
+LD_SHARED   = mpifort -shared
 AR          = ar -r
 FFTW_INC    = $(GCC_DIR)/fftw/3.3/include
 FFTW_LIB    = $(GCC_DIR)/fftw/3.3/lib
@@ -13,22 +14,26 @@ LIBXC_INC   = $(GCC_DIR)/libxc/4.0.4/inc
 LIBXC_LIB   = $(GCC_DIR)/libxc/4.0.4/lib
 LIBXSMM_INC = $(GCC_DIR)/libxsmm/1.9/include
 LIBXSMM_LIB = $(GCC_DIR)/libxsmm/1.9/lib
-DFLAGS      = -D__F2008 -D__FFTW3 -D__LIBINT -D__LIBXC -D__LIBXSMM -D__MPI_VERSION=3\
-              -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4\
+DFLAGS      = -D__F2008 -D__FFTW3 -D__LIBINT -D__LIBXC  @LIBSMM_DEFS@ -D__MPI_VERSION=3\
+              -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4\
+              -D__SCALAPACK2 -D__ELPA=201705 -D__HAS_IEEE_EXCEPTIONS\
               -D__parallel -D__SCALAPACK
 CPPFLAGS    =
-FCFLAGS     = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
-              -fopenmp -ftree-vectorize -funroll-loops -mtune=native -std=f2008\
+OPTFLAGS    = -O2 -ffast-math -ftree-vectorize -funroll-loops\
+              -fopenmp -ftree-vectorize -funroll-loops -mtune=native\
               -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC) -I$(LIBXSMM_INC)
-LDFLAGS     = $(FCFLAGS) -static
-LIBS        = $(MPI_LIBRARY_PATH)/libscalapack-gnu.a\
-              $(LIBPATH)/liblapack-gnu.a\
-              $(LIBPATH)/libblas-gnu.a\
-              $(FFTW_LIB)/libfftw3.a\
-              $(FFTW_LIB)/libfftw3_threads.a\
-              $(LIBXC_LIB)/libxcf03.a\
-              $(LIBXC_LIB)/libxc.a\
-              $(LIBINT_LIB)/libderiv.a\
-              $(LIBINT_LIB)/libint.a\
-              $(LIBXSMM_LIB)/libxsmmf.a\
-              $(LIBXSMM_LIB)/libxsmm.a
+CFLAGS      = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS    = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS     = $(DFLAGS) $(OPTFLAGS) -fopenmp -ffree-form -ffree-line-length-none -std=f2008
+LDFLAGS     = $(FCFLAGS) $(DISTLDFLAGS)
+LIBS        = -lelpa\
+              -lscalapack -lstdc++\
+              -lopenblas\
+              -lfftw3\
+              -lfftw3_omp\
+              -lxcf03\
+              -lxc\
+              -lderiv\
+              -lint\
+              @LIBSMM_LIBS@
+DATA_DIR    = /usr/share/cp2k
diff -up cp2k-6.1/arch/Linux-x86-64-gfortran.sopt.r cp2k-6.1/arch/Linux-x86-64-gfortran.sopt
--- cp2k-6.1/arch/Linux-x86-64-gfortran.sopt.r	2018-06-12 14:07:45.000000000 +0200
+++ cp2k-6.1/arch/Linux-x86-64-gfortran.sopt	2018-07-16 15:44:45.620335729 +0200
@@ -4,6 +4,7 @@ CC          = gcc
 CPP         =
 FC          = gfortran
 LD          = gfortran
+LD_SHARED   = gfortran -shared
 AR          = ar -r
 FFTW_INC    = $(GCC_DIR)/fftw/3.3/include
 FFTW_LIB    = $(GCC_DIR)/fftw/3.3/lib
@@ -13,20 +14,24 @@ LIBXC_INC   = $(GCC_DIR)/libxc/4.0.4/inc
 LIBXC_LIB   = $(GCC_DIR)/libxc/4.0.4/lib
 LIBXSMM_INC = $(GCC_DIR)/libxsmm/1.9/include
 LIBXSMM_LIB = $(GCC_DIR)/libxsmm/1.9/lib
-DFLAGS      = -D__F2008 -D__FFTW3 -D__LIBINT -D__LIBXC -D__LIBXSMM\
-              -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4
+DFLAGS      = -D__F2008 -D__FFTW3 -D__LIBINT -D__LIBXC @LIBSMM_DEFS@\
+              -D__HAS_IEEE_EXCEPTIONS\
+              -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4
 CPPFLAGS    =
 FCFLAGS     = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
-              -ftree-vectorize -funroll-loops -mtune=native -std=f2008\
+OPTFLAGS    = -O2 -ffast-math\
+              -ftree-vectorize -funroll-loops -mtune=native\
               -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC) -I$(LIBXSMM_INC)
-LDFLAGS     = $(FCFLAGS) -static
-LIBS        = $(LIBPATH)/liblapack-gnu.a\
-              $(LIBPATH)/libblas-gnu.a\
-              $(FFTW_LIB)/libfftw3.a\
-              $(LIBXC_LIB)/libxcf03.a\
-              $(LIBXC_LIB)/libxc.a\
-              $(LIBINT_LIB)/libderiv.a\
-              $(LIBINT_LIB)/libint.a\
-              $(LIBXSMM_LIB)/libxsmmf.a\
-              $(LIBXSMM_LIB)/libxsmm.a\
+CFLAGS      = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS    = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS     = $(DFLAGS) $(OPTFLAGS) -ffree-form -ffree-line-length-none -std=f2008
+LDFLAGS     = $(FCFLAGS) $(DISTLDFLAGS)
+LIBS        = -lopenblas\
+              -lfftw3\
+              -lxcf03\
+              -lxc\
+              -lderiv\
+              -lint\
+              @LIBSMM_LIBS@\
               -ldl -lpthread
+DATA_DIR    = /usr/share/cp2k
diff -up cp2k-6.1/arch/Linux-x86-64-gfortran.ssmp.r cp2k-6.1/arch/Linux-x86-64-gfortran.ssmp
--- cp2k-6.1/arch/Linux-x86-64-gfortran.ssmp.r	2018-06-12 14:07:45.000000000 +0200
+++ cp2k-6.1/arch/Linux-x86-64-gfortran.ssmp	2018-07-16 15:44:55.628427076 +0200
@@ -4,6 +4,7 @@ CC          = gcc
 CPP         =
 FC          = gfortran
 LD          = gfortran
+LD_SHARED   = gfortran -shared
 AR          = ar -r
 FFTW_INC    = $(GCC_DIR)/fftw/3.3/include
 FFTW_LIB    = $(GCC_DIR)/fftw/3.3/lib
@@ -13,21 +14,24 @@ LIBXC_INC   = $(GCC_DIR)/libxc/4.0.4/inc
 LIBXC_LIB   = $(GCC_DIR)/libxc/4.0.4/lib
 LIBXSMM_INC = $(GCC_DIR)/libxsmm/1.9/include
 LIBXSMM_LIB = $(GCC_DIR)/libxsmm/1.9/lib
-DFLAGS      = -D__F2008 -D__FFTW3 -D__LIBINT -D__LIBXC -D__LIBXSMM\
-              -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4
+DFLAGS      = -D__F2008 -D__FFTW3 -D__LIBINT -D__LIBXC @LIBSMM_DEFS@\
+              -D__HAS_IEEE_EXCEPTIONS\
+              -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4
 CPPFLAGS    =
-FCFLAGS     = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
-              -fopenmp -ftree-vectorize -funroll-loops -mtune=native -std=f2008\
+OPTFLAGS    = -O2 -ffast-math -ftree-vectorize\
+              -fopenmp -ftree-vectorize -funroll-loops -mtune=native\
               -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC) -I$(LIBXSMM_INC)
-LDFLAGS     = $(FCFLAGS) -static
-LIBS        = $(LIBPATH)/liblapack-gnu.a\
-              $(LIBPATH)/libblas-gnu.a\
-              $(FFTW_LIB)/libfftw3.a\
-              $(FFTW_LIB)/libfftw3_threads.a\
-              $(LIBXC_LIB)/libxcf03.a\
-              $(LIBXC_LIB)/libxc.a\
-              $(LIBINT_LIB)/libderiv.a\
-              $(LIBINT_LIB)/libint.a\
-              $(LIBXSMM_LIB)/libxsmmf.a\
-              $(LIBXSMM_LIB)/libxsmm.a\
+CFLAGS      = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS    = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS     = $(DFLAGS) $(OPTFLAGS) -fopenmp -ffree-form -ffree-line-length-none -std=f2008
+LDFLAGS     = $(FCFLAGS) $(DISTLDFLAGS)
+LIBS        = -lopenblas\
+              -lfftw3\
+              -lfftw3_omp\
+              -lxcf03\
+              -lxc\
+              -lderiv\
+              -lint\
+              @LIBSMM_LIBS@\
               -ldl -lpthread
+DATA_DIR    = /usr/share/cp2k
diff -up cp2k-6.1/makefiles/Makefile.r cp2k-6.1/makefiles/Makefile
--- cp2k-6.1/makefiles/Makefile.r	2018-05-24 19:16:02.000000000 +0200
+++ cp2k-6.1/makefiles/Makefile	2018-07-16 15:43:39.888735771 +0200
@@ -181,7 +181,7 @@ endif
 	$(MAKE) --version
 	@echo ""
 	@echo "========= Python ($(ONEVERSION)) ========="
-	/usr/bin/env python --version
+	/usr/bin/python3 --version
 
 else
 
@@ -451,11 +451,11 @@ FYPPFLAGS ?= -n
 
 %.o: %.F
 ifneq ($(CPP),)
-	$(TOOLSRC)/build_utils/fypp $(FYPPFLAGS) $< $*.fypped
+	fypp $(FYPPFLAGS) $< $*.fypped
 	$(CPP) $(CPPFLAGS) -D__SHORT_FILE__="\"$(subst $(SRCDIR)/,,$<)\"" -I'$(dir $<)' $*.fypped > $*.f90
 	$(FC) -c $(FCFLAGS) $*.f90 $(FCLOGPIPE)
 else
-	$(TOOLSRC)/build_utils/fypp $(FYPPFLAGS) $< $*.F90
+	fypp $(FYPPFLAGS) $< $*.F90
 	$(FC) -c $(FCFLAGS) -D__SHORT_FILE__="\"$(subst $(SRCDIR)/,,$<)\"" -I'$(dir $<)' $*.F90 $(FCLOGPIPE)
 endif
 
diff -up cp2k-6.1/tools/build_utils/check_archives.py.r cp2k-6.1/tools/build_utils/check_archives.py
--- cp2k-6.1/tools/build_utils/check_archives.py.r	2016-06-06 17:37:19.000000000 +0200
+++ cp2k-6.1/tools/build_utils/check_archives.py	2018-07-16 15:43:39.872735625 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 # author: Ole Schuett
diff -up cp2k-6.1/tools/build_utils/discover_programs.py.r cp2k-6.1/tools/build_utils/discover_programs.py
--- cp2k-6.1/tools/build_utils/discover_programs.py.r	2016-03-09 15:07:55.000000000 +0100
+++ cp2k-6.1/tools/build_utils/discover_programs.py	2018-07-16 15:43:39.872735625 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 import re, sys, os
diff -up cp2k-6.1/tools/build_utils/makedep.py.r cp2k-6.1/tools/build_utils/makedep.py
--- cp2k-6.1/tools/build_utils/makedep.py.r	2018-04-05 12:22:05.000000000 +0200
+++ cp2k-6.1/tools/build_utils/makedep.py	2018-07-16 15:43:39.873735634 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 
 import re, sys
diff -up cp2k-6.1/tools/regtesting/do_regtest.r cp2k-6.1/tools/regtesting/do_regtest
--- cp2k-6.1/tools/regtesting/do_regtest.r	2018-05-29 20:12:38.000000000 +0200
+++ cp2k-6.1/tools/regtesting/do_regtest	2018-07-16 15:43:39.874735643 +0200
@@ -263,9 +263,9 @@ cp2k_postfix=${cp2k_postfix:-"${cp2k_run
 #
 ###################################################################################
 test_types_file=${dir_base}/${cp2k_dir}/tests/TEST_TYPES
-dir_last=${dir_base}/LAST-${dir_triplet}-${cp2k_version}
+dir_last=${dir_base}/${cp2k_dir}/LAST-${dir_triplet}-${cp2k_version}
 dirout=${dirout:-${dir_base}}
-dir_out=${dir_out}/TEST-${dir_triplet}-${cp2k_version}-${datum_short}
+dir_out=${dir_base}/${cp2k_dir}/TEST-${dir_triplet}-${cp2k_version}-${datum_short}
 changelog=${dir_last}/ChangeLog
 changelog_diff=${changelog}.diff
 changelog_new=${changelog}.new
@@ -424,6 +424,7 @@ function run_regtest_dir() {
      test_tolerance=`grep -v "#" TEST_FILES | ${awk} -v itest=$itest -v def_err_tol=$default_err_tolerance '{c=c+1;if (c==itest) if (NF >= 3) { print $3 } else { print def_err_tol } }'`
      test_ref_value=`grep -v "#" TEST_FILES | ${awk} -v itest=$itest '{c=c+1;if (c==itest) if (NF == 4) { print $4 } }'`
 
+     echo "Running ${input_file}"
      output_file=${dir_out}/${dir}/${input_file}.out
      output_last=${dir_last}/${dir}/${input_file}.out
 
