diff -up cp2k-2.7.0/tools/build_utils/get_arch_code.ppc64 cp2k-2.7.0/tools/build_utils/get_arch_code
--- cp2k-2.7.0/tools/build_utils/get_arch_code.ppc64	2014-10-19 12:50:03.000000000 +0200
+++ cp2k-2.7.0/tools/build_utils/get_arch_code	2015-09-11 11:00:15.277289000 +0200
@@ -71,8 +71,6 @@
         PLATFORM="IRIX";;
       FreeBSD-*-*-*)
         PLATFORM="FreeBSD-libint-gfortran";;
-      Linux-*-*-ppc64)
-        PLATFORM="Linux-bgl-ibm";;
       Linux-*-*-alpha)
         PLATFORM="Linux-dec-alpha";;
       OSF1-*-*-alpha)
