diff -up cp2k-2.5.1/tools/get_arch_code.non-x86 cp2k-2.5.1/tools/get_arch_code
--- cp2k-2.5.1/tools/get_arch_code.non-x86	2012-12-21 21:25:02.000000000 +0100
+++ cp2k-2.5.1/tools/get_arch_code	2014-10-14 14:36:31.420908980 +0200
@@ -100,6 +100,13 @@
         else
           PLATFORM="Linux-ia64-intel"
         fi;;
+      Linux-*-*-*)
+        if [ -n "$FORT_C_NAME" ]
+        then
+          PLATFORM="Linux-${machine}-$FORT_C_NAME"
+        else
+          PLATFORM="Linux-${machine}-gfortran"
+        fi;;
       *-unicosmk-*-CRAY*T3E)
         PLATFORM="CRAY-T3E";;
       SunOS-*-sun4u)
