diff -up cp2k-6.1/src/pw/realspace_grid_cube.F.32bit cp2k-6.1/src/pw/realspace_grid_cube.F
--- cp2k-6.1/src/pw/realspace_grid_cube.F.32bit	2018-07-16 21:14:29.579258026 +0200
+++ cp2k-6.1/src/pw/realspace_grid_cube.F	2018-07-16 21:25:14.227137976 +0200
@@ -645,7 +645,7 @@ CONTAINS
       INTEGER, DIMENSION(3)                              :: lbounds, lbounds_local, ubounds, &
                                                             ubounds_local
       INTEGER, ALLOCATABLE, DIMENSION(:), TARGET         :: blocklengths
-      INTEGER(kind=file_offset), ALLOCATABLE, &
+      INTEGER(kind=address_kind), ALLOCATABLE, &
          DIMENSION(:), TARGET                            :: displacements
       INTEGER(kind=file_offset)                          :: BOF
       INTEGER                                            :: counter, gid, i, islice, j, k, last_z, &
