diff -up cp2k-4.1/arch/Linux-x86-64-gfortran.popt.r cp2k-4.1/arch/Linux-x86-64-gfortran.popt
--- cp2k-4.1/arch/Linux-x86-64-gfortran.popt.r	2017-02-08 12:17:12.913249533 +0100
+++ cp2k-4.1/arch/Linux-x86-64-gfortran.popt	2017-02-08 12:28:11.272556216 +0100
@@ -1,8 +1,9 @@
 # Tested with: GFortran 4.9.2, MPICH 3.1, LAPACK 3.5.0, ScaLAPACK 2.0.2
 CC         = gcc
 CPP        =
-FC         = mpif90
-LD         = mpif90
+FC         = mpifort
+LD         = mpifort
+LD_SHARED  = mpifort -shared
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
@@ -13,18 +14,23 @@ LIBINT_LIB = $(GCC_DIR)/libint/1.1.4-LAR
 LIBXC_INC  = $(GCC_DIR)/libxc/2.2.2-gnu/include
 LIBXC_LIB  = $(GCC_DIR)/libxc/2.2.2-gnu/lib64
 DFLAGS     = -D__FFTW3 -D__LIBINT -D__LIBXC2 -D__MPI_VERSION=3\
-             -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4\
-             -D__parallel -D__SCALAPACK
+             -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4\
+             -D__parallel -D__SCALAPACK -D__SCALAPACK2 -D__ELPA3 -D__MPI_VERSION=3
 CPPFLAGS   =
-FCFLAGS    = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
+OPTFLAGS   = -O2 -ffast-math\
              -ftree-vectorize -funroll-loops\
              -mtune=native\
              -I$(ACML_INC) -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC)
-LDFLAGS    = $(FCFLAGS) -static-libgfortran
-LIBS       = $(MPI_LIBRARY_PATH)/libscalapack-gnu.a\
-             $(ACML_LIB)/libacml.a\
-             $(FFTW_LIB)/libfftw3.a\
-             $(LIBXC_LIB)/libxcf90.a\
-             $(LIBXC_LIB)/libxc.a\
-             $(LIBINT_LIB)/libderiv.a\
-             $(LIBINT_LIB)/libint.a
+CFLAGS     = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS   = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS    = $(DFLAGS) $(OPTFLAGS) -ffree-form -ffree-line-length-none
+LDFLAGS    = $(FCFLAGS)
+LIBS       = -lelpa \
+             -lscalapack -lmpiblacsF77init -lmpiblacs -lmpiblacsCinit -lstdc++\
+             -l@BLAS@\
+             -lfftw3\
+             -lxcf90\
+             -lxc\
+             -lderiv\
+             -lint
+DATA_DIR   = /usr/share/cp2k
diff -up cp2k-4.1/arch/Linux-x86-64-gfortran.psmp.r cp2k-4.1/arch/Linux-x86-64-gfortran.psmp
--- cp2k-4.1/arch/Linux-x86-64-gfortran.psmp.r	2017-02-08 12:17:12.913249533 +0100
+++ cp2k-4.1/arch/Linux-x86-64-gfortran.psmp	2017-02-08 12:26:29.734892803 +0100
@@ -1,8 +1,9 @@
 # Tested with: GFortran 4.9.2, MPICH 3.1, LAPACK 3.5.0, ScaLAPACK 2.0.2
 CC         = gcc
 CPP        =
-FC         = mpif90
-LD         = mpif90
+FC         = mpifort
+LD         = mpifort
+LD_SHARED  = mpifort -shared
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
@@ -13,19 +14,24 @@ LIBINT_LIB = $(GCC_DIR)/libint/1.1.4-LAR
 LIBXC_INC  = $(GCC_DIR)/libxc/2.2.2-gnu/include
 LIBXC_LIB  = $(GCC_DIR)/libxc/2.2.2-gnu/lib64
 DFLAGS     = -D__FFTW3 -D__LIBINT -D__LIBXC2 -D__MPI_VERSION=3\
-             -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4\
-             -D__parallel -D__SCALAPACK
+             -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4\
+             -D__parallel -D__SCALAPACK -D__SCALAPACK2 -D__ELPA3 -D__MPI_VERSION=3
 CPPFLAGS   =
-FCFLAGS    = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
+OPTFLAGS   = -O2 -ffast-math -ftree-vectorize -funroll-loops\
              -fopenmp -ftree-vectorize -funroll-loops\
              -mtune=native\
              -I$(ACML_INC) -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC)
-LDFLAGS    = $(FCFLAGS) -static-libgfortran
-LIBS       = $(MPI_LIBRARY_PATH)/libscalapack-gnu.a\
-             $(ACML_LIB)/libacml.a\
-             $(FFTW_LIB)/libfftw3.a\
-             $(FFTW_LIB)/libfftw3_threads.a\
-             $(LIBXC_LIB)/libxcf90.a\
-             $(LIBXC_LIB)/libxc.a\
-             $(LIBINT_LIB)/libderiv.a\
-             $(LIBINT_LIB)/libint.a
+CFLAGS     = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS   = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS    = $(DFLAGS) $(OPTFLAGS) -fopenmp -ffree-form -ffree-line-length-none
+LDFLAGS    = $(FCFLAGS)
+LIBS       = -lelpa \
+             -lscalapack -lmpiblacsF77init -lmpiblacs -lmpiblacsCinit -lstdc++\
+             -l@BLAS@\
+             -lfftw3\
+             -lfftw3_omp\
+             -lxcf90\
+             -lxc\
+             -lderiv\
+             -lint
+DATA_DIR   = /usr/share/cp2k
diff -up cp2k-4.1/arch/Linux-x86-64-gfortran.sopt.r cp2k-4.1/arch/Linux-x86-64-gfortran.sopt
--- cp2k-4.1/arch/Linux-x86-64-gfortran.sopt.r	2016-07-07 10:30:49.492206000 +0200
+++ cp2k-4.1/arch/Linux-x86-64-gfortran.sopt	2017-02-08 12:28:26.892658295 +0100
@@ -3,6 +3,7 @@ CC         = gcc
 CPP        =
 FC         = gfortran
 LD         = gfortran
+LD_SHARED  = gfortran -shared
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
@@ -13,16 +14,20 @@ LIBINT_LIB = $(GCC_DIR)/libint/1.1.4-LAR
 LIBXC_INC  = $(GCC_DIR)/libxc/2.2.2-gnu/include
 LIBXC_LIB  = $(GCC_DIR)/libxc/2.2.2-gnu/lib64
 DFLAGS     = -D__FFTW3 -D__LIBINT -D__LIBXC2\
-             -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4
+             -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4
 CPPFLAGS   = 
-FCFLAGS    = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
+OPTFLAGS   = -O2 -ffast-math\
              -ftree-vectorize -funroll-loops\
              -mtune=native\
              -I$(ACML_INC) -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC)
-LDFLAGS    = $(FCFLAGS) -static-libgfortran
-LIBS       = $(ACML_LIB)/libacml.a\
-             $(FFTW_LIB)/libfftw3.a\
-             $(LIBXC_LIB)/libxcf90.a\
-             $(LIBXC_LIB)/libxc.a\
-             $(LIBINT_LIB)/libderiv.a\
-             $(LIBINT_LIB)/libint.a
+CFLAGS     = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS   = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS    = $(DFLAGS) $(OPTFLAGS) -ffree-form -ffree-line-length-none
+LDFLAGS    = $(FCFLAGS)
+LIBS       = -l@BLAS@\
+             -lfftw3\
+             -lxcf90\
+             -lxc\
+             -lderiv\
+             -lint
+DATA_DIR   = /usr/share/cp2k
diff -up cp2k-4.1/arch/Linux-x86-64-gfortran.ssmp.r cp2k-4.1/arch/Linux-x86-64-gfortran.ssmp
--- cp2k-4.1/arch/Linux-x86-64-gfortran.ssmp.r	2014-12-22 01:17:03.067670000 +0100
+++ cp2k-4.1/arch/Linux-x86-64-gfortran.ssmp	2017-02-08 12:28:31.709689775 +0100
@@ -3,6 +3,7 @@ CC         = gcc
 CPP        =
 FC         = gfortran
 LD         = gfortran
+LD_SHARED  = gfortran -shared
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
@@ -13,17 +14,20 @@ LIBINT_LIB = $(GCC_DIR)/libint/1.1.4-LAR
 LIBXC_INC  = $(GCC_DIR)/libxc/2.2.0-gnu/include
 LIBXC_LIB  = $(GCC_DIR)/libxc/2.2.0-gnu/lib64
 DFLAGS     = -D__FFTW3 -D__LIBINT -D__LIBXC2\
-             -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4
+             -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4
 CPPFLAGS   = 
-FCFLAGS    = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
-             -fopenmp -ftree-vectorize -funroll-loops\
+OPTFLAGS   = -O2 -ffast-math -ftree-vectorize -funroll-loops\
              -mtune=native\
              -I$(ACML_INC) -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC)
-LDFLAGS    = $(FCFLAGS) -static-libgfortran
-LIBS       = $(ACML_LIB)/libacml.a\
-             $(FFTW_LIB)/libfftw3.a\
-             $(FFTW_LIB)/libfftw3_threads.a\
-             $(LIBXC_LIB)/libxcf90.a\
-             $(LIBXC_LIB)/libxc.a\
-             $(LIBINT_LIB)/libderiv.a\
-             $(LIBINT_LIB)/libint.a
+CFLAGS     = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS   = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS    = $(DFLAGS) $(OPTFLAGS) -fopenmp -ffree-form -ffree-line-length-none
+LDFLAGS    = $(FCFLAGS)
+LIBS       = -l@BLAS@\
+             -lfftw3\
+             -lfftw3_omp\
+             -lxcf90\
+             -lxc\
+             -lderiv\
+             -lint
+DATA_DIR   = /usr/share/cp2k
diff -up cp2k-4.1/tools/regtesting/do_regtest.r cp2k-4.1/tools/regtesting/do_regtest
--- cp2k-4.1/tools/regtesting/do_regtest.r	2016-07-06 14:01:51.466471000 +0200
+++ cp2k-4.1/tools/regtesting/do_regtest	2017-02-08 12:17:12.914249539 +0100
@@ -247,8 +247,8 @@ cp2k_postfix=${cp2k_postfix:-"${cp2k_run
 #
 ###################################################################################
 test_types_file=${dir_base}/${cp2k_dir}/tests/TEST_TYPES
-dir_last=${dir_base}/LAST-${dir_triplet}-${cp2k_version}
-dir_out=${dir_base}/TEST-${dir_triplet}-${cp2k_version}-${datum_short}
+dir_last=${dir_base}/${cp2k_dir}/LAST-${dir_triplet}-${cp2k_version}
+dir_out=${dir_base}/${cp2k_dir}/TEST-${dir_triplet}-${cp2k_version}-${datum_short}
 changelog=${dir_last}/ChangeLog
 changelog_diff=${changelog}.diff
 changelog_new=${changelog}.new
@@ -407,6 +407,7 @@ function run_regtest_dir() {
      test_tolerance=`grep -v "#" TEST_FILES | ${awk} -v itest=$itest -v def_err_tol=$default_err_tolerance '{c=c+1;if (c==itest) if (NF >= 3) { print $3 } else { print def_err_tol } }'`
      test_ref_value=`grep -v "#" TEST_FILES | ${awk} -v itest=$itest '{c=c+1;if (c==itest) if (NF == 4) { print $4 } }'`
 
+     echo "Running ${input_file}"
      output_file=${dir_out}/${dir}/${input_file}.out
      output_last=${dir_last}/${dir}/${input_file}.out
 
