diff -up cp2k-2023.1/arch/Linux-x86-64-gfortran.psmp.r cp2k-2023.1/arch/Linux-x86-64-gfortran.psmp
--- cp2k-2023.1/arch/Linux-x86-64-gfortran.psmp.r	2023-01-02 13:56:43.266258123 +0100
+++ cp2k-2023.1/arch/Linux-x86-64-gfortran.psmp	2023-01-02 13:56:43.266258123 +0100
@@ -0,0 +1,48 @@
+# Tested with: GFortran 8.3.0, MPICH 3.3, LAPACK 3.9.1, ScaLAPACK 6819b24, FFTW 3.3.9,
+#              LIBINT 2.6.0, LIBXC 5.1.4, ELPA 2020.11.001, PLUMED 2.6.2, SPGLIB 1.16.0,
+#              LIBVORI 210412, LIBXSMM 1.16.1, SIRIUS 7.0.2, OpenBLAS 0.3.15
+# Author: Matthias Krack (matthias.krack@psi.ch, PSI, May 2021)
+
+CC          = mpicc
+FC          = mpif90
+LD          = mpif90
+LD_SHARED   = mpif90 -shared
+AR          = ar -r
+
+OPTFLAGS    = -O2 -ftree-vectorize -funroll-loops -g -march=native -mtune=native
+CFLAGS      = $(OPTFLAGS) -fopenmp -fopenmp-simd
+
+DFLAGS      = -D__ELPA
+DFLAGS     += -D__FFTW3
+DFLAGS     += -D__LIBINT
+DFLAGS     += -D__LIBXC
+DFLAGS     += @LIBSMM_DEFS@
+DFLAGS     += -D__MPI_VERSION=3
+DFLAGS     += -D__SPGLIB
+DFLAGS     += -D__MAX_CONTR=4
+DFLAGS     += -D__HAS_IEEE_EXCEPTIONS
+DFLAGS     += -D__parallel
+DFLAGS     += -D__SCALAPACK
+
+FCFLAGS     = $(CFLAGS) $(DFLAGS)
+FCFLAGS    += -fbacktrace
+FCFLAGS    += -ffree-form
+FCFLAGS    += -ffree-line-length-none
+FCFLAGS    += -fno-omit-frame-pointer
+FCFLAGS    += -std=f2008
+
+LDFLAGS     = $(FCFLAGS) $(DISTLDFLAGS)
+
+LIBS        = -lelpa_openmp
+LIBS       += -lxcf03
+LIBS       += -lxc
+LIBS       += -lint2
+LIBS       += -lsymspg
+LIBS       += -lfftw3
+LIBS       += -lfftw3_omp
+LIBS       += @LIBSMM_LIBS@
+LIBS       += -lscalapack
+LIBS       += -lflexiblas
+LIBS       += -ldl
+LIBS       += -lpthread
+LIBS       += -lstdc++
diff -up cp2k-2023.1/arch/Linux-x86-64-gfortran.ssmp.r cp2k-2023.1/arch/Linux-x86-64-gfortran.ssmp
--- cp2k-2023.1/arch/Linux-x86-64-gfortran.ssmp.r	2023-01-02 14:00:42.484316266 +0100
+++ cp2k-2023.1/arch/Linux-x86-64-gfortran.ssmp	2023-01-02 14:04:35.204397188 +0100
@@ -0,0 +1,42 @@
+# Tested with: GFortran 8.3.0, MPICH 3.3, LAPACK 3.9.1, ScaLAPACK 6819b24, FFTW 3.3.9,
+#              LIBINT 2.6.0, LIBXC 5.1.4, ELPA 2020.11.001, PLUMED 2.6.2, SPGLIB 1.16.0,
+#              LIBVORI 210412, LIBXSMM 1.16.1, SIRIUS 7.0.2, OpenBLAS 0.3.15
+# Author: Matthias Krack (matthias.krack@psi.ch, PSI, May 2021)
+
+CC          = gcc
+FC          = gfortran
+LD          = gfortran
+LD_SHARED   = gfortran -shared
+AR          = ar -r
+
+OPTFLAGS    = -O2 -ftree-vectorize -funroll-loops -g -march=native -mtune=native
+CFLAGS      = $(OPTFLAGS) -fopenmp -fopenmp-simd
+
+DFLAGS      = -D__FFTW3
+DFLAGS     += -D__LIBINT
+DFLAGS     += -D__LIBXC
+DFLAGS     += @LIBSMM_DEFS@
+DFLAGS     += -D__SPGLIB
+DFLAGS     += -D__HAS_IEEE_EXCEPTIONS
+DFLAGS     += -D__MAX_CONTR=4
+
+FCFLAGS     = $(CFLAGS) $(DFLAGS)
+FCFLAGS    += -fbacktrace
+FCFLAGS    += -ffree-form
+FCFLAGS    += -ffree-line-length-none
+FCFLAGS    += -fno-omit-frame-pointer
+FCFLAGS    += -std=f2008
+
+LDFLAGS     = $(FCFLAGS) $(DISTLDFLAGS)
+
+LIBS        = -lxcf03
+LIBS       += -lxc
+LIBS       += -lint2
+LIBS       += -lsymspg
+LIBS       += -lfftw3
+LIBS       += -lfftw3_omp
+LIBS       += @LIBSMM_LIBS@
+LIBS       += -lflexiblas
+LIBS       += -ldl
+LIBS       += -lpthread
+LIBS       += -lstdc++
diff -up cp2k-2023.1/exts/build_dbcsr/Makefile.r cp2k-2023.1/exts/build_dbcsr/Makefile
--- cp2k-2023.1/exts/build_dbcsr/Makefile.r	2022-12-31 10:53:17.602869500 +0100
+++ cp2k-2023.1/exts/build_dbcsr/Makefile	2023-01-02 13:56:43.266258123 +0100
@@ -47,7 +47,7 @@ MAKEFILE     := $(DBCSRCP2K)/Makefile
 LIBDIR       := $(DBCSRHOME)/lib
 OBJDIR       := $(DBCSRHOME)/obj
 TOOLSDIR     := $(DBCSRHOME)/tools
-FYPPEXE      := $(TOOLSDIR)/build_utils/fypp/bin/fypp
+FYPPEXE      := /usr/bin/fypp
 SRCDIR       := $(DBCSRHOME)/src
 TESTSDIR     := $(DBCSRHOME)/tests
 ARCHFILE     :=
diff -up cp2k-2023.1/Makefile.r cp2k-2023.1/Makefile
--- cp2k-2023.1/Makefile.r	2022-12-31 11:06:14.064502500 +0100
+++ cp2k-2023.1/Makefile	2023-01-02 13:56:43.267258119 +0100
@@ -16,7 +16,7 @@ export VERSION=ssmp
 
 MAKEFILE     := $(CP2KHOME)/Makefile
 ARCHDIR      := $(CP2KHOME)/arch
-DATA_DIR     := $(CP2KHOME)/data
+DATA_DIR     := /usr/share/cp2k/data
 MAINEXEDIR   := $(CP2KHOME)/exe
 MAINLIBDIR   := $(CP2KHOME)/lib
 MAINOBJDIR   := $(CP2KHOME)/obj
@@ -515,7 +515,7 @@ FCFLAGS += -D__COMPILE_ARCH="\"$(ARCH)\"
 FYPPFLAGS ?= -n
 
 %.o: %.F
-	$(TOOLSRC)/build_utils/fypp $(FYPPFLAGS) $< $*.F90
+	/usr/bin/fypp $(FYPPFLAGS) $< $*.F90
 	$(FC) -c $(FCFLAGS) -D__SHORT_FILE__="\"$(subst $(SRCDIR)/,,$<)\"" -I'$(dir $<)' $(OBJEXTSINCL) $*.F90 $(FCLOGPIPE)
 
 %.o: %.c
diff -up cp2k-2023.1/tools/regtesting/do_regtest.r cp2k-2023.1/tools/regtesting/do_regtest
--- cp2k-2023.1/tools/regtesting/do_regtest.r	2023-01-02 13:56:43.267258119 +0100
+++ cp2k-2023.1/tools/regtesting/do_regtest	2023-01-02 13:59:33.397591133 +0100
@@ -378,6 +378,7 @@ function run_regtest_dir() {
      test_tolerance=$(grep '^\s*\w' TEST_FILES | ${awk} -v itest=$itest -v def_err_tol=$default_err_tolerance '{c=c+1;if (c==itest) if (NF >= 3) { print $3 } else { print def_err_tol } }')
      test_ref_value=$(grep '^\s*\w' TEST_FILES | ${awk} -v itest=$itest '{c=c+1;if (c==itest) if (NF == 4) { print $4 } }')
 
+     echo "Running ${input_file}"
      output_file=${dir_test}/${dir}/${input_file}.out
      output_last=${dir_last}/${dir}/${input_file}.out
 
