diff -up cp2k-2.4/arch/Linux-x86-64-gfortran.popt.r cp2k-2.4/arch/Linux-x86-64-gfortran.popt
--- cp2k-2.4/arch/Linux-x86-64-gfortran.popt.r	2013-04-17 11:28:09.000000000 +0200
+++ cp2k-2.4/arch/Linux-x86-64-gfortran.popt	2013-07-02 21:46:11.102063718 +0200
@@ -12,9 +12,10 @@ LD       = mpif90
 AR       = ar -r
 
 CPPFLAGS = 
-DFLAGS   = -D__GFORTRAN -D__FFTSG -D__LIBINT -D__FFTW3 -D__parallel -D__SCALAPACK -D__BLACS -D__HAS_smm_dnn -D__LIBXC2
-FCFLAGS  = -O3 -march=native -ffast-math -funroll-loops -g -ffree-form $(DFLAGS)
-LDFLAGS  = $(FCFLAGS)  -L/data/vjoost/libint/install/lib/ -L/data/vjoost/libsmm/lib/  -L/opt/intel/composerxe-2011.3.174/mkl/lib/intel64 -L/data/vjoost/libxc-2.0.1/install/lib
-LIBS     = -lsmm_dnn -lderiv -lint -lstdc++ -lfftw3 -lmkl_gf_lp64 -lmkl_scalapack_lp64 -lmkl_blacs_openmpi_lp64 -lmkl_sequential -lmkl_core -lxc
+DFLAGS   = -D__GFORTRAN -D__FFTSG -D__LIBINT -D__FFTW3 -D__parallel -D__SCALAPACK -D__BLACS -D__LIBXC2 -D__LIBINT_MAX_AM=8 -D__LIBDERIV_MAX_AM1=6
+OPTFLAGS = -O3 -march=native -ffast-math -funroll-loops -g
+FCFLAGS  = $(OPTFLAGS) -ffree-form -ffree-line-length-none $(DFLAGS)
+LDFLAGS  = $(FCFLAGS)
+LIBS     = -lderiv -lint -lfftw3 -lstdc++ -llapack -lscalapack -lf77blas -lmpiblacsF77init -lmpiblacs -lmpiblacsCinit -lxc
 
 OBJECTS_ARCHITECTURE = machine_gfortran.o
diff -up cp2k-2.4/arch/Linux-x86-64-gfortran.psmp.r cp2k-2.4/arch/Linux-x86-64-gfortran.psmp
--- cp2k-2.4/arch/Linux-x86-64-gfortran.psmp.r	2013-04-17 11:28:09.000000000 +0200
+++ cp2k-2.4/arch/Linux-x86-64-gfortran.psmp	2013-07-02 22:05:27.872818114 +0200
@@ -13,9 +13,10 @@ LD       = mpif90
 AR       = ar -r
 
 CPPFLAGS = 
-DFLAGS   = -D__GFORTRAN -D__FFTSG -D__LIBINT -D__FFTW3 -D__parallel -D__SCALAPACK -D__BLACS -D__HAS_smm_dnn -D__LIBXC2
-FCFLAGS  = -fopenmp -O3 -march=native -ffast-math -funroll-loops -g -ffree-form $(DFLAGS)
-LDFLAGS  = $(FCFLAGS)  -L/data/vjoost/libint/install/lib/ -L/data/vjoost/libsmm/lib/  -L/opt/intel/composerxe-2011.3.174/mkl/lib/intel64 -L/data/vjoost/libxc-2.0.1/install/lib
-LIBS     = -lsmm_dnn -lderiv -lint -lstdc++ -lfftw3_threads -lfftw3 -lmkl_gf_lp64 -lmkl_scalapack_lp64 -lmkl_blacs_openmpi_lp64 -lmkl_sequential -lmkl_core -lxc
+DFLAGS   = -D__GFORTRAN -D__FFTSG -D__LIBINT -D__FFTW3 -D__parallel -D__SCALAPACK -D__BLACS -D__LIBXC2 -D__LIBINT_MAX_AM=8 -D__LIBDERIV_MAX_AM1=6
+OPTFLAGS = -fopenmp -O3 -march=native -ffast-math -funroll-loops -g
+FCFLAGS  = $(OPTFLAGS) -ffree-form -ffree-line-length-none $(DFLAGS)
+LDFLAGS  = $(FCFLAGS)
+LIBS     = -lderiv -lint -lfftw3_omp -lfftw3 -lstdc++ -llapack -lscalapack -lf77blas -lmpiblacsF77init -lmpiblacs -lmpiblacsCinit -lxc
 
 OBJECTS_ARCHITECTURE = machine_gfortran.o
diff -up cp2k-2.4/arch/Linux-x86-64-gfortran.sopt.r cp2k-2.4/arch/Linux-x86-64-gfortran.sopt
--- cp2k-2.4/arch/Linux-x86-64-gfortran.sopt.r	2013-04-17 11:28:09.000000000 +0200
+++ cp2k-2.4/arch/Linux-x86-64-gfortran.sopt	2013-06-19 21:59:08.051612836 +0200
@@ -12,9 +12,10 @@ LD       = gfortran
 AR       = ar -r
 
 CPPFLAGS = 
-DFLAGS   = -D__GFORTRAN -D__FFTSG -D__LIBINT -D__FFTW3 -D__HAS_smm_dnn -D__LIBXC2
-FCFLAGS  = -O3 -march=native -ffast-math -g -ffree-form $(DFLAGS)
-LDFLAGS  = $(FCFLAGS)  -L/data/vjoost/libint/install/lib/ -L/data/vjoost/libsmm/lib/  -L/opt/intel/composerxe-2011.3.174/mkl/lib/intel64  -L/data/vjoost/libxc-2.0.1/install/lib
-LIBS     = -lsmm_dnn -lderiv -lint -lstdc++ -lfftw3 -lmkl_gf_lp64 -lmkl_sequential -lmkl_core -lxc
+DFLAGS   = -D__GFORTRAN -D__FFTSG -D__LIBINT -D__FFTW3 -D__LIBXC2 -D__LIBINT_MAX_AM=8 -D__LIBDERIV_MAX_AM1=6
+OPTFLAGS = -O3 -march=native -ffast-math -g -ffree-form
+FCFLAGS  = $(OPTFLAGS) -ffree-form -ffree-line-length-none $(DFLAGS)
+LDFLAGS  = $(FCFLAGS)
+LIBS     = -lderiv -lint -lfftw3 -llapack -lf77blas -latlas -lxc
 
 OBJECTS_ARCHITECTURE = machine_gfortran.o
diff -up cp2k-2.4/arch/Linux-x86-64-gfortran.ssmp.r cp2k-2.4/arch/Linux-x86-64-gfortran.ssmp
--- cp2k-2.4/arch/Linux-x86-64-gfortran.ssmp.r	2013-04-17 11:28:09.000000000 +0200
+++ cp2k-2.4/arch/Linux-x86-64-gfortran.ssmp	2013-06-19 21:59:08.051612836 +0200
@@ -12,9 +12,10 @@ LD       = gfortran
 AR       = ar -r
 
 CPPFLAGS = 
-DFLAGS   = -D__GFORTRAN -D__FFTSG -D__LIBINT -D__FFTW3 -D__HAS_smm_dnn -D__LIBXC2
-FCFLAGS  = -fopenmp -O3 -march=native -ffast-math -g -ffree-form $(DFLAGS)
-LDFLAGS  = $(FCFLAGS)  -L/data/vjoost/libint/install/lib/ -L/data/vjoost/libsmm/lib/  -L/opt/intel/composerxe-2011.3.174/mkl/lib/intel64  -L/data/vjoost/libxc-2.0.1/install/lib
-LIBS     = -lsmm_dnn -lderiv -lint -lstdc++ -lfftw3 -lmkl_gf_lp64 -lmkl_sequential -lmkl_core -lxc
+DFLAGS   = -D__GFORTRAN -D__FFTSG -D__LIBINT -D__FFTW3 -D__LIBXC2 -D__LIBINT_MAX_AM=8 -D__LIBDERIV_MAX_AM1=6
+OPTFLAGS = -O3 -march=native -ffast-math -g
+FCFLAGS  = $(OPTFLAGS) -ffree-form -ffree-line-length-none -fopenmp $(DFLAGS)
+LDFLAGS  = $(FCFLAGS) 
+LIBS     = -lderiv -lint -lfftw3_omp -lfftw3 -llapack -lf77blas -latlas -lxc
 
 OBJECTS_ARCHITECTURE = machine_gfortran.o
diff -up cp2k-2.4/makefiles/Makefile.r cp2k-2.4/makefiles/Makefile
--- cp2k-2.4/makefiles/Makefile.r	2013-05-27 13:12:21.000000000 +0200
+++ cp2k-2.4/makefiles/Makefile	2013-06-19 21:59:08.051612836 +0200
@@ -96,17 +96,17 @@ CLEAN_TARGETS = $(addsuffix /clean ,$(VE
 # lists the targets that are not files
 #
 .PHONY : _all libs _progr _lib build clean realclean distclean $(CLEAN_TARGETS)\
-         $(ALL_TARGETS) $(LIB_TARGETS) $(VERSION_TARGETS) fes dirs deptool\
+         $(ALL_TARGETS) $(LIB_TARGETS) $(VERSION_TARGETS) fes dirs\
          dbcsr_test dbcsr_perf mrproper plumed          
 
 ### Master rules ###
-build: dirs deptool
+build: dirs
 	+$(MAKE) -C $(SRCDIR) -f $(MAKEFILE) all.dep
 	+cp $(LIBCUDA_DEPENDENCIES) $(OBJDIR)/$(VERSION)/libc_cuda.dep
 	+cp $(LIBMA_DEPENDENCIES) $(OBJDIR)/$(VERSION)/libc_ma.dep
 	+$(MAKE) -C $(OBJDIR)/$(VERSION) -f $(MAKEFILE) $(WHAT)
 
-fes: dirs deptool
+fes: dirs
 	+$(MAKE) -C $(TOOLSRC)/metadyn/  -f $(MAKEFILE) fes.dep
 	+$(MAKE) -C $(OBJDIR)/$(VERSION) -f $(MAKEFILE) HACKLIBDEP=yes VERSION=$(VERSION) $(FES) 
 
@@ -127,9 +127,6 @@ pretty: dirs
 	+$(MAKE) -C $(SRCDIR)/arch_lib  -f $(MAKEFILE) _pretty_lib4
 	+$(MAKE) -C $(SRCDIR)/elpa_lib  -f $(MAKEFILE) _pretty_lib5
 
-deptool: dirs
-	+$(MAKE) LDFLAGS=$(MAKDEPF90_LDFLAGS) -C $(TOOLDIR) -f $(MAKEFILE) makedepf90
-
 libs:
 	+$(MAKE) -f $(MAKEFILE) VERSION=$(VERSION) WHAT=_lib build
 all:
@@ -154,7 +151,7 @@ plumedobj: dirs
 #
 # on a parallel build only one task should go a build the deptool
 #
-$(VERSION_TARGETS): dirs deptool
+$(VERSION_TARGETS): dirs
 	+$(MAKE) -f $(MAKEFILE) VERSION=$@ build
 
 $(CLEAN_TARGETS):
@@ -269,13 +266,13 @@ ifeq ($(HACKDEP),yes)
 	echo $(ALL_SOURCE_FILES) | awk '{for (i=1;i<=NF;i++) {obj=$$(i) ; sub(".F",".o",obj); if ($$(i)!=obj) print obj,":",$$(i) }}' > $(OBJDIR)/$(VERSION)/all.dep
 else
    ifeq ($(MC),)
-	$(TOOLDIR)/makedepf90 -I $(SRCDIRS) $(MODSTRING) -free $(ALL_SOURCE_FILES) > $(OBJDIR)/$(VERSION)/all.dep || ( rm -f $(OBJDIR)/$(VERSION)/all.dep  ; exit 1 )
+	makedepf90 -I $(SRCDIRS) $(MODSTRING) -free $(ALL_SOURCE_FILES) > $(OBJDIR)/$(VERSION)/all.dep || ( rm -f $(OBJDIR)/$(VERSION)/all.dep  ; exit 1 )
    else
 # if MC is defined, it is our 'module compiler' which generates the .mod file from the source file
 # it is useful in a two-stage compile. Therefore we split the dependency in two parts the .o and the .mod.
 # the awk below assumes our current rule of one mod per file
 # The current module compiler is 'gfortran -fsyntax-only ' which should work fine for gfortran 4.8.0 and higher.
-	$(TOOLDIR)/makedepf90 -I $(SRCDIRS) $(MODSTRING) -free $(ALL_SOURCE_FILES) | \
+	makedepf90 -I $(SRCDIRS) $(MODSTRING) -free $(ALL_SOURCE_FILES) | \
             awk '{deps=substr($$0,index($$0,":"),length($$0)-index($$0,":")+1); if ($$2==":") { print $$1,deps } else {print $$2,deps ; print $$1,deps,$$2 }}' > \
             $(OBJDIR)/$(VERSION)/all.dep || ( rm -f $(OBJDIR)/$(VERSION)/all.dep  ; exit 1 )
    endif
@@ -283,7 +280,7 @@ endif
 
 # this is kind of a hack, fes depends on files in other directories, hence the -I and the -b.
 fes.dep: $(FESOBJS:.o=.F) 
-	$(TOOLDIR)/makedepf90 -I$(SRCDIR) -b. $(MODSTRING)  -free $^ > $(OBJDIR)/$(VERSION)/fes.dep || ( rm -f $(OBJDIR)/$(VERSION)/fes.dep  ; exit 1 )
+	makedepf90 -I$(SRCDIR) -b. $(MODSTRING)  -free $^ > $(OBJDIR)/$(VERSION)/fes.dep || ( rm -f $(OBJDIR)/$(VERSION)/fes.dep  ; exit 1 )
 
 plumed.dep: $(OBJECTS:.o=.F) $(LIB1_OBJECTS:.o=.F) $(LIB2_OBJECTS:.o=.F) $(LIB4_OBJECTS:.o=.F) $(LIB3_OBJECTS:.o=.F) $(LIB5_OBJECTS:.o=.F) cp_common_uses.h src-plumed/metadyn.h cp2k.F 
 
@@ -412,9 +409,6 @@ dbcsr_performance_driver.o: $(LIB3_ARCHI
 #
 # these are tools, but their integration in the build is ugly at least (e.g. see dependencies).
 #
-makedepf90:
-	-test -d $(TOOLDIR)/makedepf90-build || ( mkdir makedepf90-build ; cp $(TOOLSRC)/makedepf90/* makedepf90-build )
-	{ cd makedepf90-build ; ./configure --prefix=$(TOOLDIR) --bindir=$(TOOLDIR) ; $(MAKE) VERSION="2.8.8cp2k" ; $(MAKE) install ; }
 
 #
 # the rule how to generate the .o from the .F
diff -up cp2k-2.4/tools/do_regtest.r cp2k-2.4/tools/do_regtest
--- cp2k-2.4/tools/do_regtest.r	2012-08-21 10:25:41.000000000 +0200
+++ cp2k-2.4/tools/do_regtest	2013-06-19 21:59:08.051612836 +0200
@@ -103,8 +103,8 @@ leakcheck="NO"
 
 # *** how to execute an input file [ cp2k_prefix input cp2k_postfix ]
 # Leave empty for serial, uncomment for parallel
-cp2k_run_prefix="mpiexec -np 2"
-#cp2k_run_prefix=""
+#cp2k_run_prefix="mpiexec -np 2"
+cp2k_run_prefix=""
 cp2k_run_postfix=
 #cp2k_prefix="( ulimit -t 300; ${dir_base}/${cp2k_dir}/exe/${dir_triplet}/cp2k.${cp2k_version}"
 #cp2k_postfix=" )"
@@ -353,9 +353,6 @@ if [[ ${quick} != "quick" ]]; then
    else
       echo "make realclean VERSION=${cp2k_version} went fine"
    fi
-else
-  echo "quick testing, no realclean"
-fi 
 
 # *** from here failures are likely to be bugs in cp2k
 echo "-------------------------compiling cp2k-----------------------------------"
@@ -372,6 +369,10 @@ else
 compile_warnings=`grep "Warning:" out | wc | tail -1 |  ${awk} '{print $1}'`
 echo "make VERSION=${cp2k_version} went fine (${compile_warnings} warnings)"
 fi
+
+else
+  echo "quick testing, no realclean"
+fi 
 echo "-------------------------regtesting cp2k----------------------------------"
 
 ###################################################################################
@@ -519,6 +520,7 @@ do
      input_file=`grep -v "#" TEST_FILES | ${awk} -v itest=$itest '{c=c+1;if (c==itest) print $1}'`
      # just one test right now, but this should generalize
      test_types=`grep -v "#" TEST_FILES | ${awk} -v itest=$itest '{c=c+1;if (c==itest) print $2}'`
+     echo "Running ${input_file}"
      output_file=${dir_out}/${dir}/${input_file}.out
      output_last=${dir_last}/${dir}/${input_file}.out
      ( ulimit -t ${job_max_time} -v 2000000 ; ${cp2k_prefix} ${input_file} ${cp2k_postfix} &> ${output_file} )
