diff -up cp2k-4.1/src/xc/xc_libxc_wrap.F.libxc3 cp2k-4.1/src/xc/xc_libxc_wrap.F
--- cp2k-4.1/src/xc/xc_libxc_wrap.F.libxc3	2017-02-08 12:32:43.881337121 +0100
+++ cp2k-4.1/src/xc/xc_libxc_wrap.F	2017-02-08 12:36:04.341644838 +0100
@@ -58,14 +58,19 @@ MODULE xc_libxc_wrap
                                              XC_HYB_GGA_XC_O3LYP,&
                                              XC_HYB_GGA_XC_X3LYP,&
                                              XC_HYB_GGA_X_N12_SX,&
-                                             XC_HYB_MGGA_X_M11,&
                                              XC_LDA_C_1D_CSC,&
                                              XC_LDA_C_2D_PRM,&
                                              XC_LDA_C_XALPHA,&
                                              XC_LDA_X,&
                                              XC_LDA_X_1D,&
                                              XC_MGGA_X_M11_L,&
+#if ( XC_MAJOR_VERSION == 2)
+                                             XC_HYB_MGGA_X_M11,&
                                              XC_MGGA_X_MS2H,&
+#else
+                                             XC_HYB_MGGA_XC_M11,&
+                                             XC_HYB_MGGA_X_MS2H,&
+#endif
                                              XC_MGGA_X_TB09
 #else
    ! Functionals which require parameters
