diff -up cp2k-2.7.0/arch/Linux-x86-64-gfortran.popt.r cp2k-2.7.0/arch/Linux-x86-64-gfortran.popt
--- cp2k-2.7.0/arch/Linux-x86-64-gfortran.popt.r	2014-10-30 09:11:05.000000000 +0100
+++ cp2k-2.7.0/arch/Linux-x86-64-gfortran.popt	2015-09-06 00:22:14.361166707 +0200
@@ -1,8 +1,8 @@
 # Tested with: GFortran 4.9.1, MPICH 3.1, LAPACK 3.5.0, ScaLAPACK 2.0.2
 CC         = gcc
 CPP        =
-FC         = mpif90
-LD         = mpif90
+FC         = mpifort
+LD         = mpifort
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
@@ -13,18 +13,22 @@ LIBINT_LIB = $(GCC_DIR)/libint/1.1.4-LAR
 LIBXC_INC  = $(GCC_DIR)/libxc/2.2.0-gnu/include
 LIBXC_LIB  = $(GCC_DIR)/libxc/2.2.0-gnu/lib64
 DFLAGS     = -D__FFTW3 -D__LIBINT -D__LIBXC2\
-             -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4\
-             -D__parallel -D__SCALAPACK
+             -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4\
+             -D__parallel -D__SCALAPACK -D__SCALAPACK2 -D__ELPA2 -D__MPI_VERSION=3
 CPPFLAGS   =
-FCFLAGS    = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
+OPTFLAGS   = -O2 -ffast-math\
              -ftree-vectorize -funroll-loops\
              -mtune=native\
              -I$(ACML_INC) -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC)
-LDFLAGS    = $(FCFLAGS) -static-libgfortran
-LIBS       = $(MPI_LIBRARY_PATH)/libscalapack-gnu.a\
-             $(ACML_LIB)/libacml.a\
-             $(FFTW_LIB)/libfftw3.a\
-             $(LIBXC_LIB)/libxcf90.a\
-             $(LIBXC_LIB)/libxc.a\
-             $(LIBINT_LIB)/libderiv.a\
-             $(LIBINT_LIB)/libint.a
+CFLAGS     = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS   = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS    = $(DFLAGS) $(OPTFLAGS) -ffree-form -ffree-line-length-none
+LDFLAGS    = $(FCFLAGS)
+LIBS       = -lelpa \
+             -lscalapack -lmpiblacsF77init -lmpiblacs -lmpiblacsCinit -lstdc++\
+             -lsatlas\
+             -lfftw3\
+             -lxc\
+             -lderiv\
+             -lint
+DATA_DIR   = /usr/share/cp2k
diff -up cp2k-2.7.0/arch/Linux-x86-64-gfortran.psmp.r cp2k-2.7.0/arch/Linux-x86-64-gfortran.psmp
--- cp2k-2.7.0/arch/Linux-x86-64-gfortran.psmp.r	2014-12-22 01:17:03.000000000 +0100
+++ cp2k-2.7.0/arch/Linux-x86-64-gfortran.psmp	2015-09-06 00:22:10.299198948 +0200
@@ -1,8 +1,8 @@
 # Tested with: GFortran 4.9.1, MPICH 3.1, LAPACK 3.5.0, ScaLAPACK 2.0.2
 CC         = gcc
 CPP        =
-FC         = mpif90
-LD         = mpif90
+FC         = mpifort
+LD         = mpifort
 AR         = ar -r
 ACML_INC   = $(ACMLPATH)/include
 ACML_LIB   = $(ACMLPATH)/lib
@@ -13,19 +13,22 @@ LIBINT_LIB = $(GCC_DIR)/libint/1.1.4-LAR
 LIBXC_INC  = $(GCC_DIR)/libxc/2.2.0-gnu/include
 LIBXC_LIB  = $(GCC_DIR)/libxc/2.2.0-gnu/lib64
 DFLAGS     = -D__FFTW3 -D__LIBINT -D__LIBXC2\
-             -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4\
-             -D__parallel -D__SCALAPACK
+             -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4\
+             -D__parallel -D__SCALAPACK -D__SCALAPACK2 -D__ELPA2 -D__MPI_VERSION=3
 CPPFLAGS   =
-FCFLAGS    = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
-             -fopenmp -ftree-vectorize -funroll-loops\
+OPTFLAGS   = -O2 -ffast-math -ftree-vectorize -funroll-loops\
              -mtune=native\
              -I$(ACML_INC) -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC)
-LDFLAGS    = $(FCFLAGS) -static-libgfortran
-LIBS       = $(MPI_LIBRARY_PATH)/libscalapack-gnu.a\
-             $(ACML_LIB)/libacml.a\
-             $(FFTW_LIB)/libfftw3.a\
-             $(FFTW_LIB)/libfftw3_threads.a\
-             $(LIBXC_LIB)/libxcf90.a\
-             $(LIBXC_LIB)/libxc.a\
-             $(LIBINT_LIB)/libderiv.a\
-             $(LIBINT_LIB)/libint.a
+CFLAGS     = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS   = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS    = $(DFLAGS) $(OPTFLAGS) -fopenmp -ffree-form -ffree-line-length-none
+LDFLAGS    = $(FCFLAGS)
+LIBS       = -lelpa \
+             -lscalapack -lmpiblacsF77init -lmpiblacs -lmpiblacsCinit -lstdc++\
+             -lsatlas\
+             -lfftw3\
+             -lfftw3_omp\
+             -lxc\
+             -lderiv\
+             -lint
+DATA_DIR   = /usr/share/cp2k
diff -up cp2k-2.7.0/arch/Linux-x86-64-gfortran.sopt.r cp2k-2.7.0/arch/Linux-x86-64-gfortran.sopt
--- cp2k-2.7.0/arch/Linux-x86-64-gfortran.sopt.r	2014-12-22 01:17:03.000000000 +0100
+++ cp2k-2.7.0/arch/Linux-x86-64-gfortran.sopt	2015-09-06 00:21:48.178374523 +0200
@@ -13,17 +13,19 @@ LIBINT_LIB = $(GCC_DIR)/libint/1.1.4-LAR
 LIBXC_INC  = $(GCC_DIR)/libxc/2.2.0-gnu/include
 LIBXC_LIB  = $(GCC_DIR)/libxc/2.2.0-gnu/lib64
 DFLAGS     = -D__FFTW3 -D__LIBINT -D__LIBXC2\
-             -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4
+             -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4
 CPPFLAGS   = 
-FCFLAGS    = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
+OPTFLAGS   = -O2 -ffast-math\
              -ftree-vectorize -funroll-loops\
              -mtune=native\
              -I$(ACML_INC) -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC)
-LDFLAGS    = $(FCFLAGS) -static-libgfortran
-LIBS       = $(ACML_LIB)/libacml.a\
-             $(FFTW_LIB)/libfftw3.a\
-             $(LIBXC_LIB)/libxcf90.a\
-             $(LIBXC_LIB)/libxc.a\
-             $(LIBINT_LIB)/libderiv.a\
-             $(LIBINT_LIB)/libint.a
-
+CFLAGS     = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS   = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS    = $(DFLAGS) $(OPTFLAGS) -ffree-form -ffree-line-length-none
+LDFLAGS    = $(FCFLAGS)
+LIBS       = -lsatlas\
+             -lfftw3\
+             -lxc\
+             -lderiv\
+             -lint
+DATA_DIR   = /usr/share/cp2k
diff -up cp2k-2.7.0/arch/Linux-x86-64-gfortran.ssmp.r cp2k-2.7.0/arch/Linux-x86-64-gfortran.ssmp
--- cp2k-2.7.0/arch/Linux-x86-64-gfortran.ssmp.r	2014-12-22 01:17:03.000000000 +0100
+++ cp2k-2.7.0/arch/Linux-x86-64-gfortran.ssmp	2015-09-06 00:22:00.241278778 +0200
@@ -13,17 +13,19 @@ LIBINT_LIB = $(GCC_DIR)/libint/1.1.4-LAR
 LIBXC_INC  = $(GCC_DIR)/libxc/2.2.0-gnu/include
 LIBXC_LIB  = $(GCC_DIR)/libxc/2.2.0-gnu/lib64
 DFLAGS     = -D__FFTW3 -D__LIBINT -D__LIBXC2\
-             -D__LIBINT_MAX_AM=7 -D__LIBDERIV_MAX_AM1=6 -D__MAX_CONTR=4
+             -D__LIBINT_MAX_AM=@LIBINT_MAX_AM@ -D__LIBDERIV_MAX_AM1=@LIBDERIV_MAX_AM@ -D__MAX_CONTR=4
 CPPFLAGS   = 
-FCFLAGS    = $(DFLAGS) -O2 -ffast-math -ffree-form -ffree-line-length-none\
-             -fopenmp -ftree-vectorize -funroll-loops\
+OPTFLAGS   = -O2 -ffast-math -ftree-vectorize -funroll-loops\
              -mtune=native\
              -I$(ACML_INC) -I$(FFTW_INC) -I$(LIBINT_INC) -I$(LIBXC_INC)
-LDFLAGS    = $(FCFLAGS) -static-libgfortran
-LIBS       = $(ACML_LIB)/libacml.a\
-             $(FFTW_LIB)/libfftw3.a\
-             $(FFTW_LIB)/libfftw3_threads.a\
-             $(LIBXC_LIB)/libxcf90.a\
-             $(LIBXC_LIB)/libxc.a\
-             $(LIBINT_LIB)/libderiv.a\
-             $(LIBINT_LIB)/libint.a
+CFLAGS     = $(DFLAGS) $(OPTFLAGS)
+CXXFLAGS   = $(DFLAGS) $(OPTFLAGS)
+FCFLAGS    = $(DFLAGS) $(OPTFLAGS) -fopenmp -ffree-form -ffree-line-length-none
+LDFLAGS    = $(FCFLAGS)
+LIBS       = -lsatlas\
+             -lfftw3\
+             -lfftw3_omp\
+             -lxc\
+             -lderiv\
+             -lint
+DATA_DIR   = /usr/share/cp2k
diff -up cp2k-2.7.0/tools/regtesting/do_regtest.r cp2k-2.7.0/tools/regtesting/do_regtest
--- cp2k-2.7.0/tools/regtesting/do_regtest.r	2015-09-01 09:12:22.000000000 +0200
+++ cp2k-2.7.0/tools/regtesting/do_regtest	2015-09-06 00:20:17.995096835 +0200
@@ -229,8 +229,8 @@ cp2k_postfix=${cp2k_postfix:-"${cp2k_run
 #
 ###################################################################################
 test_types_file=${dir_base}/${cp2k_dir}/tests/TEST_TYPES
-dir_last=${dir_base}/LAST-${dir_triplet}-${cp2k_version}
-dir_out=${dir_base}/TEST-${dir_triplet}-${cp2k_version}-${datum_short}
+dir_last=${dir_base}/${cp2k_dir}/LAST-${dir_triplet}-${cp2k_version}
+dir_out=${dir_base}/${cp2k_dir}/TEST-${dir_triplet}-${cp2k_version}-${datum_short}
 changelog=${dir_last}/ChangeLog
 changelog_diff=${changelog}.diff
 changelog_new=${changelog}.new
@@ -389,6 +389,7 @@ function run_regtest_dir() {
      test_tolerance=`grep -v "#" TEST_FILES | ${awk} -v itest=$itest -v def_err_tol=$default_err_tolerance '{c=c+1;if (c==itest) if (NF >= 3) { print $3 } else { print def_err_tol } }'`
      test_ref_value=`grep -v "#" TEST_FILES | ${awk} -v itest=$itest '{c=c+1;if (c==itest) if (NF == 4) { print $4 } }'`
 
+     echo "Running ${input_file}"
      output_file=${dir_out}/${dir}/${input_file}.out
      output_last=${dir_last}/${dir}/${input_file}.out
 
