diff -up cp2k-3.0/src/xc/xc_libxc_wrap.F.libxc212 cp2k-3.0/src/xc/xc_libxc_wrap.F
--- cp2k-3.0/src/xc/xc_libxc_wrap.F.libxc212	2015-10-21 19:18:15.000000000 +0200
+++ cp2k-3.0/src/xc/xc_libxc_wrap.F	2016-04-11 14:22:54.933855120 +0200
@@ -27,8 +27,8 @@ MODULE xc_libxc_wrap
 #if defined (__LIBXC)
 #include <xc_version.h>
 ! check for LibXC version
-#if (XC_MAJOR_VERSION < 2) || ((XC_MAJOR_VERSION == 2) && (XC_MINOR_VERSION < 2))
-  This version of CP2K ONLY works with libxc versions 2.2.2 and above.
+#if (XC_MAJOR_VERSION < 2) || ((XC_MAJOR_VERSION == 2) && (XC_MINOR_VERSION < 1))
+  This version of CP2K ONLY works with libxc versions 2.1.2 and above.
   Furthermore, -I[LIBXC_DIR]/INCLUDE needs to be added to FCFLAGS.
 #else
 
